include/zephyr/sys/util.h

```diff
@@ -639,6 +639,45 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
 	(((buflen) != 0) &&                        \
 	((UINTPTR_MAX - (uintptr_t)(addr)) <= ((uintptr_t)((buflen) - 1))))
 
+/**
+ * @brief XOR n bytes
+ *
+ * @param dst  Destination of where to store result. Shall be @p len bytes.
+ * @param src1 First source. Shall be @p len bytes.
+ * @param src2 Second source. Shall be @p len bytes.
+ * @param len  Number of bytes to XOR.
+ */
+static inline void mem_xor_n(uint8_t *dst, const uint8_t *src1, const uint8_t *src2, size_t len)
+{
+	while (len--) {
+		*dst++ = *src1++ ^ *src2++;
+	}
+}
+
+/**
+ * @brief XOR 32 bits
+ *
+ * @param dst  Destination of where to store result. Shall be 32 bits.
+ * @param src1 First source. Shall be 32 bits.
+ * @param src2 Second source. Shall be 32 bits.
+ */
+static inline void mem_xor_32(uint8_t dst[4], const uint8_t src1[4], const uint8_t src2[4])
+{
+	mem_xor_n(dst, src1, src2, 4U);
```

Comment:
One possible optimization here would be to cast to uint32_t and then xor directly.

Reply:
Not a bad suggestion. Since the code was copied from implementations from @cvinayak I'd like his input as to why it was not done like that in the first place.

Reply:
This is to support CPUs needing 4-byte aligned memory read/write operations (ARM cortex-m0 variants etc...).

Fun fact, was my very first contribution to Zephyr Project! https://github.com/zephyrproject-rtos/zephyr/commit/de999a81c58b842becea1e7003fc35bd7d3ff789

Reply:
Probably solvable by using `UNALIGNED_PUT/UNALIGNED_GET` macros.

Reply:
Sorry look at your comment on UNALIGNED_PUT/UNALIGNED_GET now. The implementation in our toolchain header has workarounds for compiler version, and my feeling goes towards sticking to simple old while loop for now.